### DISCUSSION TITLE
Stopping and resuming indexing on all nodes

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#15188] [Q&A] [answered]
I am trying to define a startup process for our system where I:
1) Stop indexing 
2) Make a very heavy modifications to existing data patching over and over again many documents (that it why I do not want indexing to hijack resources till I am finished). 
3) Resume indexing

_store.Maintenance.Send(new StopIndexingOperation()) stops indexing only in one node
_store.Maintenance.Send(new DisableIndexOperation(index.Name, true)) seems to behave inconsistently - after Enabling _store.Maintenance.Send(new EnableIndexOperation(index.Name, true)) some indexes remain on some nodes in disabled state. 

Any recommendation, how I could pause indexing on all nodes for the time when I am planning to execute step 2 of my process and resume is consistently on all nodes, once I am done.

**Comment by garayx** [comment, marked as answer]
Hi,

You have to read all the indexes names and then disable them one at a time, so you using the `DisableIndexOperation(index.Name, true)` operation is the right way to go. Make sure you are passing the true flag so the Enable/Disable operation is cluster wide.
